contracts/staking/ServiceStakingBase.sol

```diff
@@ -435,20 +440,81 @@ abstract contract ServiceStakingBase is ERC721TokenReceiver, IErrorsRegistries {
         }
     }
 
+    /// @dev Evicts services due to their extended inactivity.
+    /// @param evictServiceIds Service Ids to be evicted.
+    function _evict(uint256[] memory evictServiceIds) internal {
+        uint256 totalNumServices = evictServiceIds.length;
+        uint256 numEvictServices;
+
+        // Get the number of evicted services
+        for (uint256 i = 0; i < totalNumServices; ++i) {
+            if (evictServiceIds[i] > 0) {
+                numEvictServices++;
+            }
+        }
+
+        if (numEvictServices > 0) {
+            // Get arrays of exact sizes
+            uint256[] memory serviceIds = new uint256[](numEvictServices);
+            address[] memory owners = new address[](numEvictServices);
+            address[] memory multisigs = new address[](numEvictServices);
+            uint256[] memory serviceInactivity = new uint256[](numEvictServices);
+            uint256[] memory serviceIndexes = new uint256[](numEvictServices);
+
+            // Fill in arrays
+            uint256 sCounter;
+            uint256 serviceId;
+            for (uint256 i = 0; i < totalNumServices; ++i) {
+                if (evictServiceIds[i] > 0) {
+                    serviceId = evictServiceIds[i];
+                    serviceIds[sCounter] = serviceId;
+
+                    ServiceInfo storage sInfo = mapServiceInfo[serviceId];
+                    owners[sCounter] = sInfo.owner;
+                    multisigs[sCounter] = sInfo.multisig;
+                    serviceInactivity[sCounter] = sInfo.inactivity;
+                    serviceIndexes[sCounter] = i;
+                    sCounter++;
+                }
+            }
+
+            // Evict services from the global set of staked services
+            uint256 idx;
+            for (uint256 i = numEvictServices - 1; i > 0; --i) {
```

Comment:
When evicting, we go backwards, as if we go forward all the indexes will be changed with every deletion.